### DISCUSSION TITLE
Overlay 3D markers on camera images in Image panel

### DISCUSSION THREAD

**Discussion by esthersweon** [foxglove/community#158] [Q&A]
It should be possible to overlay 3D markers (currently rendered only in the 3D panel) onto a 2D image in the Image panel.

**Comment by amacneil** [comment]
We're currently in the process of a ground up rewrite of the 3d panel. That project is a prerequisite for rendering 3D markers into the image panel. So it's definitely on our roadmap, but probably not for a couple months. In the meantime, the workaround is to calculate the [ImageMarker](https://foxglove.dev/docs/studio/panels/image#imageannotation) positions yourself and publish them as a separate topic.

**Comment by gilduz** [comment]
Hi all, is there any news about this topic? Including rendering of point cloud on top of 2D images.

**Comment by amacneil** [comment]
We just launched the new 3D panel out of beta this week, so it will need some time for that to bake and fix any issues that come up. We do not currently have a planned timeline for rendering 3d objects or point clouds in the image panel.

**Comment by gilduz** [comment]
Thanks for the update. I tried new 3D panel and it's great. The ability to
render 3D objects on image panel will be ultimate feature to definitely
switch to foxglove.

**Comment by adamconkey** [comment]
This is the main feature that's in rviz that I definitely want in Foxglove. In particular I'd love to be able to have all meshes and coordinate frames overlayed on the image. Seems like another way to achieve this is to include your camera feed in the 3D view, set an alpha on the camera image so it's semi-transparent, and align the 3D viewer view to exactly match the camera perspective. I haven't figured out a clean way to achieve that though.

**Reply by tanzeyy** [reply]
Hi, since it has been a while, have you found out how to render 3D annotations over the camera image panel?

**Reply by jtbandes** [reply]
We are actively working on this feature. Stay tuned!

**Comment by gilduz** [comment]
Super great!!!!

**Comment by jtbandes** [comment]
We’ve just pushed out the first beta release of the new image panel! Go to studio.foxglove.dev and try it out — the panel is named “Image (Beta)”.

**Reply by tanzeyy** [reply]
Great! Is there any document on how to use this new panel?

**Reply by snosenzo** [reply]
> Great! Is there any document on how to use this new panel?

Unfortunately, not yet. However much of the 3D capabilities should be the same as documented in the 3D panel. Outside of that it should be only a matter of choosing your image and calibration topics.
